### PR TITLE
Superficial kube-proxy cleanup

### PR DESCRIPTION
**Pull request by vllry**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
I did some surface-level cleanup while reading the kube-proxy code. This PR has simple changes like:

* Renaming variables that collide with imported packages
* Fiddly Go best-practices, like ensuring error strings are all lowercase and without punctuation
* TODO notes on issues

**Which issue(s) this PR fixes**:
None

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: cmd/kube-proxy/app/server_others_test.go [hunk 1]
```diff
@@ -67,7 +67,7 @@ func Test_getProxyMode(t *testing.T) {
 		},
 		{ // detect, error
 			flag:          "",
-			iptablesError: fmt.Errorf("oops!"),
+			iptablesError: fmt.Errorf("oops"),
```
Comment (lavalamp) [inline cmd/kube-proxy/app/server_others_test.go]: I like to use errors.New if there's no actual arguments to the error; but it doesn't really matter much and this is an error just for the test anyway.

Comment (vllry) [reply cmd/kube-proxy/app/server_others_test.go]: I wasn't sure enough about best-practices there to suggest a change, but I usually do that too.

**Pull request merged**